tests/Behat/Content/Node/ProjectContentContext.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content\Behat\Content\Node;
+
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+use Drupal\Tests\oe_content\Behat\Hook\Scope\BeforeParseEntityFieldsScope;
+
+/**
+ * Context to create project content entities.
+ */
+class ProjectContentContext extends RawDrupalContext {
+
+  /**
+   * Run before fields are parsed by Drupal Behat extension.
+   *
+   * @param \Drupal\Tests\oe_content\Behat\Hook\Scope\BeforeParseEntityFieldsScope $scope
+   *   Behat hook scope.
+   *
+   * @BeforeParseEntityFields(node,oe_project)
+   */
+  public function alterProjectFields(BeforeParseEntityFieldsScope $scope): void {
+    // Map human readable field names to their Behat parsable machine names.
+    $mapping = [
+      'Title' => 'title',
+      'Summary' => 'oe_summary',
+      'Reference' => 'oe_reference',
+      'Project period' => 'oe_project_dates',
+      'Overall budget' => 'oe_project_budget',
+      'EU contribution' => 'oe_project_budget_eu',
+      'Website' => 'oe_project_website',
+      'Body text' => 'body',
+      'Call for proposals' => 'oe_project_calls',
+      'Results' => 'oe_project_results',
+      'Result files' => 'oe_project_result_files',
+      'Alternative title' => 'oe_content_short_title',
+      'Navigation title' => 'oe_content_navigation_title',
+      'Teaser' => 'oe_teaser',
+      'Departments' => 'oe_project_departments',
+      'Coordinators' => 'oe_project_coordinators',
+      'Funding programme' => 'oe_project_funding_programme',
+      'Participants' => 'oe_project_participants',
+      'Published' => 'status',
```

Comment:
Many of this fields will need further processing in order for them to be usable in a Behat test, such as `oe_project_departments` that references a SKOS vocabulary.

For fields such as that, on the event context, we have something like:

```php
        case 'Type':
        case 'Languages':
        case 'Internal organiser':
          $fields = $this->getRevisionField($mapping[$key], 'skos_concept', $value);
          $scope->addFields($fields)->removeField($key);
          break;
```

So we either complete each one of the fields above properly, by also adding a test that proves that this works, or we only add here the minimum, such as "title", "published" and other simple value fields (text, float, long text, etc.).

Or we can simply remove this context all together, as it's not used in this PR. We could move this in the theme ticket, as a companion PR on the oe_content, as this context will be used to test the project theming.

Up to you, only if we ship with this, then let's make sure that what we ship is complete and working properly, albeit minimal.

Reply:
done